core/order.go

```diff
@@ -549,6 +571,19 @@ func (n *OpenBazaarNode) createContractWithOrder(data *PurchaseData) (*pb.Ricard
 			return nil, fmt.Errorf("Contract only accepts %s, our wallet uses %s", listing.Metadata.AcceptedCurrencies[0], n.Wallet.CurrencyCode())
 		}
 
+		// Make sure shipping fields are filled if the order contains a physical good
+		if containsPhysicalGood && !(n.TestNetworkEnabled() || n.RegressionNetworkEnabled()) {
```

Comment:
I would like you to stop using this pattern:
```
var thing bool
for {
  if condition {
    thing = true
  }
}
```

The mutation is difficult to reason about and creates unexpected bugs when you forget certain cases (which are spread over a 50-100 lines of code). Instead, these variables want to be a function which accepts the dependancies of that condition.

`containsPhysicalGood(sl)`
```
func containsPhysicalGoods(sl) bool {
  if sl.Listing.Metadata.ContractType == pb.Listing_Metadata_PHYSICAL_GOOD {
    return true
  }
  return false
}

Reply:
Or even better:
`sl.containsPhysicalGood()`